tests/test_calc_resid.py

```diff
@@ -0,0 +1,37 @@
+"""
+This file tests the calculate residual helper function, used in .fit().
```

Comment:
@agupta01 The tests are correctly formatted for pytest, everything should work. Wasn't sure if you were debugging the tests so added some comments to help with that.